api/scpca_portal/models/dataset.py

```diff
@@ -87,12 +87,12 @@ def get_or_find_ccdl_dataset(
         if dataset := cls.objects.filter(
             is_ccdl=True, ccdl_name=ccdl_name, ccdl_project_id=project_id
         ).first():
-            return dataset, False
+            return dataset, True
 
         dataset = cls(is_ccdl=True, ccdl_name=ccdl_name, ccdl_project_id=project_id)
         dataset.format = dataset.ccdl_type["format"]
         dataset.data = dataset.get_ccdl_data()
-        return dataset, True
+        return dataset, False
```

Comment:
Suggestion: Can we utilize Django built-in [`.get_or_create`](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#get-or-create) shortcut method to optimize the logic by using the `created` boolean flag?

>  Returns a tuple of `(object, created),` where object is the retrieved or created object and created is a boolean specifying whether a new object was created.

Reply:
We had discussed this a bit before landing on the current implementation. The difference is that we don't want to return a saved version of the dataset. Django doesn't allow for us to override that behavior with their `get_or_create` convenience method.

It would be nice if there was an existing django convention / convenience method that would allow for returning either a retrieved instance or a new unsaved instance. If a better option becomes available switching it over wouldn't be that difficult so for now I think we can keep it this way.